### PR TITLE
Add authentication note to API docs and clarify auth content in API overview

### LINKED ISSUE #2410: Add API auth note at the top of each API doc
### Description
We should add a note at the top of each API doc for authenticated APIs that explains where to find more information about authenticating for API access with a link to authentication info in API Overview.

We should also look through the API auth content and update/clarify as needed. For example, the API auth content should mention `sensuctl env`.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [x] improvement (improve docs content or format)

### Affected pages
- All API docs except health and metrics
- https://docs.sensu.io/sensu-go/latest/api/overview/#access-control

### Expected behavior
API docs include cURL examples with an authorization command that refers to the Sensu access token, so users should be able to find more information about how to authenticate from the API doc pages.

### Current behavior
API docs include cURL examples with an authorization command that refers to the Sensu access token, but none link to auth info in the API overview.

### Corrections needed
- Add note at the top of every API doc page (except health and metrics) that links to auth info in API Overview
- Review API Overview auth content and update or clarify as needed.
- Add `sensuctl env` info and link in API Overview

### LINKED ISSUE #2497: API examples should illustrate use of keys for authentication
### Description

Our API examples currently illustrate the use of access tokens, but we want to make API keys the recommended practice for integrating with Sensu.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages

All API reference docs

### Expected behavior

As a developer I can provision an API key and `export SENSU_API_KEY=....` to have copy/paste success with Sensu API examples

### Current behavior

As a developer I can run `eval $(sensuctl env)` periodically to refresh `$SENSU_ACCESS_TOKEN` so I can have copy/paste success with Sensu API examples.

### Additional context

API keys are preferable to access tokens because they can be administratively revoked and don't need to be refreshed periodically.

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Adds note about authentication requirements at the top of each API doc (currently only APIKey doc; see review instructions note below)
- Clarifies authentication content in the API overview doc and adds a link to `sensuctl env` content in the sensuctl reference doc

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2410
Closes https://github.com/sensu/sensu-docs/issues/2497

## Review Instructions
I can propagate the note to the docs for all APIs that require access token or API key authentication when I have it 👌 

Also, I'm wondering if the code in https://docs.sensu.io/sensu-go/latest/api/overview/#authentication-quickstart should align a little better with https://docs.sensu.io/sensu-go/latest/sensuctl/reference/#usage-1. I'm not sure I understand why they are different. Does the code in the API overview quickstart call the auth API and set the env var all at the same time?

### CODE REVIEW

**Comment by cwjohnston** [comment]
> Also, I'm wondering if the code in https://docs.sensu.io/sensu-go/latest/api/overview/#authentication-quickstart should align a little better with https://docs.sensu.io/sensu-go/latest/sensuctl/reference/#usage-1. I'm not sure I understand why they are different. Does the code in the API overview quickstart call the auth API and set the env var all at the same time?

The approach in the API overview quickstart is useful because it doesn't rely on the sensuctl binary, although it does rely on `jq` to make it a one-liner:

```
# Requires curl and jq
export SENSU_USER=YOUR_USERNAME && SENSU_PASS=YOUR_PASSWORD
export SENSU_ACCESS_TOKEN=`curl -X GET -u "$SENSU_USER:$SENSU_PASS" -s http://localhost:8080/auth | jq -r ".access_token"`
```

The backticks around the curl command are analogous to the `$()` syntax used by the `eval $(sensuctl env)` command from the sensuctl reference.

I don't think the sensuctl reference makes it clear, but there's no need to run `sensuctl env` before running `eval $(sensuctl env)`. 

The latter `eval` command is the true one-liner to set a number of useful environment variables, provided sensuctl has already been configured. Under the hood, `sensuctl env` is calling the same /auth API endpoint as needed.

All that said, the `eval $(sensuctl env)` incantation is convenient for manual API calls when working with Sensu on the command line, but not a recommended practice for integrating with Sensu. Likewise, the management of access and refresh tokens is somewhat burdensome for developers, compared to API keys which do not require periodic refresh.

I think we should work toward making API keys the default authentication method for our examples, so I opened https://github.com/sensu/sensu-docs/issues/2497.

**Review by cwjohnston — APPROVED** [review]
LGTM!

**Comment by hillaryfraley** [comment]
Thank you for this explanation, @cwjohnston . I think I will update this PR to also address https://github.com/sensu/sensu-docs/issues/2497 -- this PR will require touching every API doc, so I might as well make the API key changes while I'm at it. Thanks again.

**Comment by hillaryfraley** [comment]
@cwjohnston I'm sorry to ask you to review this again, but I did a little bit of rewriting in the API overview doc to accommodate the change to using API keys in the API doc examples.

File: content/sensu-go/5.20/api/overview.md [hunk 1]
```diff
@@ -170,7 +167,39 @@ API keys are cluster-wide resources, so only cluster admins can grant, view, and
...
+
+Code examples in the Sensu API docs use the environment variable `$SENSU_API_KEY` to represent a valid API key in API requests.
+
+Use [sensuctl][18] or the [APIkeys API][19] to generate an API key.
+Then, follow this example to export your API key to a custom `SENSU_API_KEY` environment variable you can use for API authentication:
+
+{{< language-toggle >}}
+
+{{< highlight bash >}}
+sensuctl env
```
Comment (cwjohnston) [inline content/sensu-go/5.20/api/overview.md]:
```suggestion
```

File: content/sensu-go/5.20/api/overview.md [hunk 2]
```diff
@@ -170,7 +167,39 @@ API keys are cluster-wide resources, so only cluster admins can grant, view, and
...
+export SENSU_API_KEY="83abef1e-e7d7-4beb-91fc-79ad90084d5b"
+
+# Run this command to configure your shell:
+# eval $(sensuctl env)
+{{< /highlight >}}
+
+{{< highlight cmd >}}
+sensuctl env --shell cmd
```
Comment (cwjohnston) [inline content/sensu-go/5.20/api/overview.md]:
```suggestion
```

File: content/sensu-go/5.20/api/overview.md [hunk 3]
```diff
@@ -170,7 +167,39 @@ API keys are cluster-wide resources, so only cluster admins can grant, view, and
...
+SET SENSU_API_KEY="83abef1e-e7d7-4beb-91fc-79ad90084d5b"
+{{< /highlight >}}
+
+{{< highlight powershell >}}
+sensuctl env --shell powershell
```
Comment (cwjohnston) [inline content/sensu-go/5.20/api/overview.md]:
```suggestion
```

File: content/sensu-go/5.20/api/overview.md [hunk 4]
```diff
@@ -170,7 +167,39 @@ API keys are cluster-wide resources, so only cluster admins can grant, view, and
...
+$Env:SENSU_API_KEY = "83abef1e-e7d7-4beb-91fc-79ad90084d5b"
+
+# Run this command to configure your shell:
+# & sensuctl env --shell powershell | Invoke-Expression
```
Comment (cwjohnston) [inline content/sensu-go/5.20/api/overview.md]:
```suggestion
```

File: content/sensu-go/5.20/api/overview.md [hunk 5]
```diff
@@ -170,7 +167,39 @@ API keys are cluster-wide resources, so only cluster admins can grant, view, and
...
+Then, follow this example to export your API key to a custom `SENSU_API_KEY` environment variable you can use for API authentication:
+
+{{< language-toggle >}}
+
+{{< highlight bash >}}
+sensuctl env
+export SENSU_API_KEY="83abef1e-e7d7-4beb-91fc-79ad90084d5b"
+
+# Run this command to configure your shell:
+# eval $(sensuctl env)
```
Comment (cwjohnston) [inline content/sensu-go/5.20/api/overview.md]:
```suggestion
```

File: content/sensu-go/5.20/api/overview.md [hunk 6]
```diff
@@ -170,7 +167,39 @@ API keys are cluster-wide resources, so only cluster admins can grant, view, and
 **NOTE**: API keys are not supported for authentication providers such as LDAP and OIDC.
 {{% /notice %}}
 
-#### API key authentication
+#### Configure an environment variable for API key authentication
+
+Code examples in the Sensu API docs use the environment variable `$SENSU_API_KEY` to represent a valid API key in API requests.
+
+Use [sensuctl][18] or the [APIkeys API][19] to generate an API key.
+Then, follow this example to export your API key to a custom `SENSU_API_KEY` environment variable you can use for API authentication:
```
Comment (cwjohnston) [inline content/sensu-go/5.20/api/overview.md]:
```suggestion
Then, follow this example to export your API key to the `SENSU_API_KEY` environment variable you can use for API authentication:
```

**Review by cwjohnston — CHANGES_REQUESTED** [review]

File: content/sensu-go/5.20/api/overview.md [hunk 7]
```diff
@@ -170,7 +167,32 @@ API keys are cluster-wide resources, so only cluster admins can grant, view, and
...
+
+{{< /highlight >}}
+
+{{< highlight cmd >}}
+SET SENSU_API_KEY="83abef1e-e7d7-4beb-91fc-79ad90084d5b"
+{{< /highlight >}}
+
+{{< highlight powershell >}}
+$Env:SENSU_API_KEY = "83abef1e-e7d7-4beb-91fc-79ad90084d5b"
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.20/api/overview.md]:
```suggestion
```

File: content/sensu-go/5.20/api/overview.md [hunk 8]
```diff
@@ -170,7 +167,31 @@ API keys are cluster-wide resources, so only cluster admins can grant, view, and
...
+Code examples in the Sensu API docs use the environment variable `$SENSU_API_KEY` to represent a valid API key in API requests.
+
+Use [sensuctl][18] or the [APIkeys API][19] to generate an API key.
+Then, follow this example to export your API key to the `SENSU_API_KEY` environment variable you can use for API authentication:
+
+{{< language-toggle >}}
+
+{{< highlight bash >}}
+export SENSU_API_KEY="83abef1e-e7d7-4beb-91fc-79ad90084d5b"
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.20/api/overview.md]:
```suggestion
```

**Pull request merged**